msteams-platform/concepts/messaging-extensions/create-extensions.md

```diff
@@ -339,22 +352,130 @@ This used to insert a card into the compose box as a result of a the command. It
+
+1. The user clicks the messaging extension to trigger the task module.
+1. The user uses the task module to configure the poll.
+1. After submitting the configuration task module the app uses the information provided in the task module to craft an adaptive card and sends it as a `botMessagePreview` response to the client.
+1. The user can then preview the adaptive card message before the bot will inserts it into the channel. If the bot is not already a member of the channel, clicking `Send` will add the bot.
+1. Interacting with the adaptive card will change the message before sending it.
+1. Once the user clicks `Send` the bot will post the message to the channel.
+
+To enable this flow your task module should respond as in the example below, which will present the preview message to the user.
+
+```json
+{
+  "composeExtension": {
+    "type": "botMessagePreview",
+    "activityPreview": {
+      "attachments":  [
+        {
+          "contentType": "application/vnd.microsoft.card.adaptive",
+          "content": << Card Payload >>
+        }
+      ]
+    }
+  }
+}
+```
+
+Your message extension will now need to respond to two new types of interactions, `value.botMessagePreviewAction = "send"` and `value.botMessagePreviewAction = "edit"`. Below is an example of the `value` object you will need to process:
+
+```json
+{
+  "name": "composeExtension/submitAction",
+  "type": "invoke",
+  ...
+  "value": {
+    "botMessagePreviewAction": "send",
+    "botActivityPreview": [
+      {
+        "type": "message/card",
+        "attachments": [
+          {
+            "content":
               {
-                "url": "https://placekitten.com/200/200"
+                "type": "AdaptiveCard",
+                "body": [{<<card payload>>}]
               }
-            ]
           }
-        }
+        ],
+        "context": { "theme": "default" }
       }
     ],
-    "type": "result",
-    "attachmentLayout": "list"
+    "conversation": { "id": "19:c366b75791784100b6e8b515fd55b063@thread.skype" },
+    "name": "composeExtension/submitAction",
+    "imdisplayname": "Pranav Smith"
   }
 }
 ```
+
+When responding to the `edit` request you should respond with an adaptive card similar to the original `fetch\task` response, but with the values populated with the information the user has already submitted. When responding to the `submit` request you should send a message to the channel containing the finalized adaptive card. The example below shows how to do this using the [Node.js Teams Bot Builder SDK](https://www.npmjs.com/package/botbuilder-teams).
```

Comment:
@clearab this description is not valid anymore. We had design flaw and now the correct design & description is that - since message preview is always part of task module, so the response of 'edit' must be `task` payload (and this is what @aosolis  expected about). Please revise the statement that the bot response to the 'edit' invoke must follow the "TaskInfo" (named 'task') payload schema, so it could continue with a card or a webview in next frame. I'm posting an example below.